src/poetry/core/constraints/version/version_range.py

```diff
@@ -58,13 +58,25 @@ def is_any(self) -> bool:
     def is_simple(self) -> bool:
         return self._min is None or self._max is None
 
+    @property
+    def is_inclusive(self) -> bool:
+        return self._include_min or self.include_max
+
+    @property
+    def is_exclusive(self) -> bool:
+        return not self.is_inclusive
+
     def allows(self, other: Version) -> bool:
         if self._min is not None:
             _this, _other = self.allowed_min, other
 
             assert _this is not None
 
-            if not _this.is_postrelease() and _other.is_postrelease():
+            if (
+                self.is_exclusive
```

Comment:
```suggestion
                not self._include_min
```

That should be sufficient concerning the comment below, shouldn't it? (`<V` will not allow a post-release anyway.)